bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageData.java

```diff
@@ -239,6 +240,84 @@ public final class ImageData implements Cloneable {
 		{ 0x000000, 0x800000, 0x200000, 0xa00000, 0x080000, 0x880000, 0x280000, 0xa80000 }
 	};
 
+	/**
+	 * Creates an {@code ImageData} loaded from the specified file. Throws an error
+	 * if an error occurs loading the image, or if the image has an unsupported
+	 * type.
+	 * <p>
+	 * This factory is provided for convenience when loading a single image only. If
+	 * the file contains multiple images, only the first one will be loaded. To load
+	 * multiple images, use {@code ImageLoader.load()}
+	 * </p>
+	 *
+	 * @param file the file to load the image from (must not be null)
+	 *
+	 * @exception IllegalArgumentException <ul>
+	 *    <li>ERROR_NULL_ARGUMENT - if the file name is null</li>
+	 * </ul>
+	 * @exception SWTException <ul>
+	 *    <li>ERROR_IO - if an IO error occurs while reading from the file</li>
+	 *    <li>ERROR_INVALID_IMAGE - if the image file contains invalid data</li>
+	 *    <li>ERROR_UNSUPPORTED_FORMAT - if the image file contains an unrecognized format</li>
+	 * </ul>
+	 * @since 3.129
+	 */
+	public static ImageData load(Path file) {
```

Comment:
Instead of `load()` this method could also be named `create()`, but the former seems a bit more suitable for me.